File: Archived/BruteForce/6603.py
import sys

init = list(map(int,sys.stdin.readline().split()))
def recur(idx,cnt):
    global K,S,used
    if cnt == 6:
        tmp = list()
        for i in range(K):
            if used[i]:
                tmp.append(str(S[i]))
        print(' '.join(tmp))
    
    for i in range(idx,K):
        if used[i]:
            continue
        used[i] = True
        recur(i,cnt+1)
        used[i] = False
    
while init[0]:
    K = init[0]; S = init[1:]
    used = list(False for i in range(K))
    recur(0,0)
    print()
    init = list(map(int,sys.stdin.readline().split()))

File: Archived/BruteForce/6603.java
import java.io.*;
import java.util.*;

public class Main{
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    static int K;
    static int[] S;
    static StringBuilder sb = new StringBuilder(100000);
    public static void main(String[] args) throws IOException{
        int[] tmp = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
        while(tmp[0]!=0){
            K = tmp[0];
            S = Arrays.copyOfRange(tmp,1,tmp.length);
            recur(-1,0, new ArrayList<Integer>(100000));
            sb.append('\n');
            tmp = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
        }
        bw.write(sb.toString());
        bw.flush();bw.close();br.close();
    }
    public static void recur(int idx, int cnt, ArrayList<Integer> arr){
        if(cnt==6){
            for(int i: arr)
                sb.append(i).append(' ');
            sb.append('\n');
            return;
        }
        for(int i=idx+1;i<S.length;i++){
            arr.add(S[i]);
            recur(i,cnt+1,arr);
            arr.remove(arr.size()-1);
        }
    }
}